### PR TITLE
UP-4819: Add JavaScript based linters to CI process

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] tests are included

##### Description of change

[UP-4819](https://issues.jasig.org/browse/UP-4819)

Adds three new lint tasks: `npm_run_lint-js`, `npm_run_lint-less`, and `npm_run_lint-md`.
Runs all three lint tasks as part of continuous integration.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Build is failing on lint error as expected https://travis-ci.org/Jasig/uPortal/builds/215658298#L6134

**Comment by bjagg** [comment]
@ChristianMurphy can you address the conflicts? Let's get this merged in the next few days.

File: .travis.yml [hunk 1]
```diff
@@ -5,14 +5,15 @@ sudo: false
 before_install:
   - unset GEM_PATH
 
-# ./gradlew assemble (default install command) won't cut it
 install:
-  - ./gradlew install
+  - ./gradlew install # ./gradlew assemble (default install command) won't cut it
+  - ./gradlew npm_install
```
Comment (ChristianMurphy) [inline .travis.yml]: TODO: Break each linter into its own NPM script so they can be run independently

Comment (ChristianMurphy) [inline .travis.yml]: @apetro @vertein tasks have been broken into independent scripts.

**Comment by ChristianMurphy** [comment]
Brace yourselves, linter is coming

**Comment by ChristianMurphy** [comment]
This PR has grown stale, I'm going to close this out, then (hopefully) open 3-4 smaller PRs bringing the linters in.

**Pull request closed without merging**